nibabel/parrec.py

```diff
@@ -1008,13 +1085,18 @@ def get_sorted_slice_indices(self):
             array, and (equivalently) the only dimension of
             ``self.image_defs``.
         """
-        slice_nos = self.image_defs['slice number']
-        is_full = vol_is_full(slice_nos, self.general_info['max_slices'])
-        keys = (slice_nos, vol_numbers(slice_nos), np.logical_not(is_full))
-        # Figure out how many we need to remove from the end, and trim them
-        # Based on our sorting, they should always be last
+        if not self.strict_sort:
```

Comment:
Maybe move into `_lax_sort_keys method` and do `keys = self._strict_sort_keys() if self.strict_sort else self._lax_sort_keys()`